src/librustc_resolve/lib.rs

```diff
@@ -3275,8 +3278,10 @@ impl<'a> Resolver<'a> {
                             // This was an attempt to use a type parameter outside
                             // its scope.
                             if record_used {
+                                debug!("rib: {:?}", rib);
                                 resolve_error(self, span,
-                                              ResolutionError::TypeParametersFromOuterFunction);
+                                    ResolutionError::TypeParametersFromOuterFunction(None));
```

Comment:
OK, so, here `def.def_id()` gives you the `DefId` of the thing we want -- annoyingly, I'm not sure how to get the span from that. Normally I'd say to use the hir-map, but it's not available in resolve. Perhaps @eddyb or @petrochenkov can answer that -- what is the best way to get the span where `def` was declared here?